Change Allocator logs to reference tasks instead of generic values

# Description

We need to change the Allocator logging to identify each message, if we launch multiple instances, messages usually use `instance_id` or VirtualBox `UUID` references, we need to log the specific task instead (system-version-arch-id)

```
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance i-0430c13acd72fbe83 deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance i-0430c13acd72fbe83 deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance i-0430c13acd72fbe83 created successfully.
```

Should be something like

```
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance ubuntu-20-amd64-1234 deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance ubuntu-20-amd64-1234 deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR: Instance ubuntu-20-amd64-1234 created successfully.
```

or better

```
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR [ubuntu-20-amd64-1234]: Instance i-0430c13acd72fbe83 deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR [ubuntu-20-amd64-1234]: Instance i-0430c13acd72fbe83  deleted.
[2024-07-25 17:27:41,993] [INFO] ALLOCATOR [ubuntu-20-amd64-1234]: Instancei-0430c13acd72fbe83  created successfully.
```

We also need to review the resource delete-create messages, as the below message does not make sense

We need to change this so we are able to check correctly the release tests

## Related 

- https://github.com/wazuh/wazuh/issues/24935

The delete-create message bug is generated in the case of rollback:

```console
[2024-07-26 12:19:48] [WARNING] ALLOCATOR: Error on attempt 30 of 30: [Errno None] Unable to connect to port 2200 on 75.101.226.184
[2024-07-26 12:20:18] [ERROR] ALLOCATOR: Connection attempts failed after 30 tries. Connection timeout.
[2024-07-26 12:20:18] [WARNING] ALLOCATOR: Rolling back instance creation.
[2024-07-26 12:20:19] [DEBUG] ALLOCATOR: Deleting credentials: cbordon-test-key-4830
[2024-07-26 12:20:51] [INFO] ALLOCATOR: Instance i-0b12b4135c41fd539 deleted.
[2024-07-26 12:20:51] [INFO] ALLOCATOR: Instance i-0b12b4135c41fd539 deleted.
[2024-07-26 12:20:51] [INFO] ALLOCATOR: Instance i-0b12b4135c41fd539 created successfully.

```

--- a/deployability/modules/allocation/allocation.py
+++ b/deployability/modules/allocation/allocation.py
@@ -13,7 +13,7 @@ import winrm
 
 from .aws.provider import AWSProvider, AWSConfig
 from .generic import Instance, Provider, models
-from .generic.utils import logger
+from .generic.utils import logger, logger_with_instance_name
 from .vagrant.provider import VagrantProvider, VagrantConfig
 
 
@@ -58,6 +58,9 @@ class Allocator:
         config = cls.___get_custom_config(payload)
         instance = provider.create_instance(
             payload.working_dir, instance_params, config, payload.ssh_key)
+
+        global logger
+        logger = logger_with_instance_name(instance)
         logger.info(f"Instance {instance.identifier} created.")
         # Start the instance.
         instance.start()
@@ -74,10 +77,10 @@ class Allocator:
                 logger.warning(f"Rolling back instance creation.")
                 track_payload = {'track_output': track_file}
                 cls.__delete(track_payload)
-                logger.info(f"Instance {instance.identifier} deleted.")
             else:
                 logger.warning(f'The VM will not be automatically removed. Please remove it executing Allocation module with --action delete.')
-        logger.info(f"Instance {instance.identifier} created successfully.")
+        else:
+            logger.info(f"Instance {instance.identifier} created successfully.")
 
     @classmethod
     def __delete(cls, payload: models.InstancePayload) -> None:
@@ -94,6 +97,9 @@ class Allocator:
             track = models.TrackOutput(**yaml.safe_load(f))
         provider = PROVIDERS[track.provider]()
         provider.destroy_instance(models.InstancePayload(**dict(track)))
+
+        global logger
+        logger = logger_with_instance_name(track)
         logger.info(f"Instance {track.identifier} deleted.")
 
     @staticmethod
@@ -176,6 +182,7 @@ class Allocator:
             inventory = models.InventoryOutput(**yaml.safe_load(f))
         port = inventory.ansible_port
         track = models.TrackOutput(identifier=instance.identifier,
+                                    name=instance.name,
                                     provider=provider_name,
                                     instance_dir=str(instance.path),
                                     key_path=str(instance.credentials.key_path),

--- a/deployability/modules/allocation/aws/provider.py
+++ b/deployability/modules/allocation/aws/provider.py
@@ -12,7 +12,7 @@ from datetime import datetime, timedelta
 import subprocess
 
 from modules.allocation.generic import Provider
-from modules.allocation.generic.models import CreationPayload, InstancePayload, InstancePayload
+from modules.allocation.generic.models import CreationPayload, InstancePayload
 from modules.allocation.generic.utils import logger
 from .credentials import AWSCredentials
 from .instance import AWSInstance
@@ -129,6 +129,7 @@ class AWSProvider(Provider):
 
         instance_params = {}
         instance_params['instance_dir'] = instance_dir
+        instance_params['name'] = config.name
         instance_params['identifier'] = instance_id
         instance_params['platform'] = platform
         instance_params['host_identifier'] = host_identifier

--- a/deployability/modules/allocation/generic/instance.py
+++ b/deployability/modules/allocation/generic/instance.py
@@ -4,7 +4,6 @@
 
 from abc import ABC, abstractmethod
 from pathlib import Path
-from .utils import logger
 
 from .credentials import Credentials
 from .models import ConnectionInfo, InstancePayload
@@ -43,6 +42,7 @@ class Instance(ABC):
 
         self.path: Path = path
         self.identifier: str = str(instance_parameters.identifier)
+        self.name: str = instance_parameters.name
         self.credentials: Credentials = credentials
         self.host_identifier: str = instance_parameters.host_identifier
         self.host_instance_dir: Path = instance_parameters.host_instance_dir

--- a/deployability/modules/allocation/generic/models.py
+++ b/deployability/modules/allocation/generic/models.py
@@ -36,6 +36,7 @@ class InventoryOutput(BaseModel):
 
 class TrackOutput(BaseModel):
     identifier: str
+    name: str
     provider: str
     instance_dir: str
     key_path: str
@@ -116,6 +117,7 @@ class TrackPayload(BaseModel):
 
 class InstancePayload(BaseModel):
     identifier: str
+    name: str | None = None
     instance_dir: str | Path
     key_path: Path | None = None
     host_identifier: str  | None = None

--- a/deployability/modules/allocation/generic/utils.py
+++ b/deployability/modules/allocation/generic/utils.py
@@ -2,6 +2,27 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
+import logging
+from modules.allocation.generic.instance import Instance
+from modules.allocation.generic.models import TrackOutput
 from modules.generic.logger import Logger
 
+# Default logger
 logger = Logger("allocator").get_logger()
+
+
+def logger_with_instance_name(instance_info: Instance | TrackOutput) -> logging.Logger:
+    """
+    Returns a logger with the instance name if it is different from the identifier,
+    otherwise returns the default logger without the name.
+
+    Args:
+        instance (Instance): The instance object.
+
+    Returns:
+        logging.Logger: The logger object.
+    """
+
+    if instance_info.name != instance_info.identifier:
+        return Logger(f"allocator [{instance_info.name}]").get_logger()
+    return logger

--- a/deployability/modules/allocation/vagrant/provider.py
+++ b/deployability/modules/allocation/vagrant/provider.py
@@ -121,6 +121,7 @@ class VagrantProvider(Provider):
         instance_params = {}
         instance_params['instance_dir'] = instance_dir
         instance_params['identifier'] = instance_id
+        instance_params['name'] = config.name
         instance_params['platform'] = platform
         instance_params['host_identifier'] = host_identifier
         instance_params['host_instance_dir'] = host_instance_dir

